### PR TITLE
feat: generic `DataFrame`

### LINKED ISSUE #292: generic signature for dataframes
## Motivation

continuation of #151.

to reiterate:

- The type `DataFrame` in `nodejs-polars` lack generic type signature.
- Being able to know dataframe's structure in the type signature improves productivity and safety.
- This can be achieved using typescript's powerful type system. (e.g: [`Pick`](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys), [`Omit`](https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys))

## Feature Request

- Add generic type parameter for `DataFrame` type
- Defaults to `any` for type checker performance
- (Later) make methods of `DataFrame` and `Series` dependently typed.

### To-be

```ts
import pl from "npm:nodejs-polars"

const df = pl.DataFrame({
	fruit: ["Apples", "Oranges"],
	comparability: [0, 1],
})
// df: pl.DataFrame<{ fruit: pl.Series<"Utf8">, comparability: pl.Series<"Float64"> }>

const obj = df.toObject()
// obj: { fruit: string[], comparability: number[] }

const fruit = df.getColumn('fruit')
// fruit: pl.Series<"Float64">
```

## Additional Notes

current progress (2024-11-14):

### PR DESCRIPTION
**Pull request by scarf005**
image

resolves #292. please check individual commits for detail.

### CODE REVIEW

File: __tests__/dataframe.test.ts [hunk 1]
```diff
@@ -1314,8 +1314,8 @@ describe("dataframe", () => {
     ]);
     expect(actual).toFrameEqual(expected);
   });
-  test("pivot", () => {
```
Comment (Bidek56) [inline __tests__/dataframe.test.ts:1317]: Any reason for removing this? Thx

File: __tests__/dataframe.test.ts [hunk 2]
```diff
@@ -1335,82 +1335,88 @@ describe("dataframe", () => {
...
       on: "a",
       aggregateFunc: "first",
       sortColumns: true,
     });
 
     expect(actual).toFrameEqual(expected, true);
+  }
 
-    df = pl.DataFrame({
+  {
```
Comment (Bidek56) [inline __tests__/dataframe.test.ts:1348]: Any reason for these extra `{ }` ? thx

File: __tests__/dataframe.test.ts (hunk 1 above)
Comment (scarf005) [reply __tests__/dataframe.test.ts:1317]: gah, accidently removed when adding scopes, will amend and push again

**Review by Bidek56 — APPROVED** [review]

**Comment by scarf005** [comment]
> Overall looks good, but I think we should update some of the docstrings to include examples with the new type system.

applied in [`ed11909` (#293)](https://github.com/pola-rs/nodejs-polars/pull/293/commits/ed119097fa76be1bdc5aa293642cd35404075c66) and [`1cf2275` (#293)](https://github.com/pola-rs/nodejs-polars/pull/293/commits/1cf227500c92a11b5035ed8cafe71f2f72ba6c54).

**Review by universalmind303 — APPROVED** [review]

**Comment by universalmind303** [comment]
thanks @scarf005

**Pull request merged**